packages/formik/src/hooks/useFullFormikState.ts

```diff
@@ -0,0 +1,21 @@
+import { useMemo } from 'react';
+import { FormikApi, FormikComputedState, FormikState } from '../types';
+import { selectComputedState } from './useFormikComputedState';
+
+export const selectFullState = <T>(value: T) => value;
+
+export const useFullFormikState = <Values>(
```

Comment:
full, computed, api, state, selectFull, need to reduce the surface area.

Reply:
- `useFormikApi` is needed, as it gets the Context, which is now only the API
- `useFormikState` is an alias to `(selector, comparer) => useFormikApi().useState(selector, comparer)`, which seemed a little tedious to make users use. However, we have to expose that for any of this to work, so it's up to you whether we keep this hook or make users use `useFormikApi().useState()`
- `useFormikApiState(api, ...etc)` can probably go, since it is just `api.useState(selector, comparer)`;
- `selectFullState` is not something we should expose, it's just something we use internally instead of memoizing a new selector. I can go through and mark stuff like this @internal.
- `useFullFormikState` is used by backwards compatible `useFormikContext` which users expect to load the  legacy "formikBag". It's also used internally to subscribe to the full state in the case of Formik and Field `render` / `child fn`s which do not have access to hooks to get Formik's state.

Reply:
Can useFormikState become useFormikStateSelector() or just useFormikSelector()?

I would also like to suggest changing useFormikApi() to useFormikActions() or useFormikMethods(). Mentally, easier to think about state or actions/methods versus state or api I think. api could be anything and it isn’t clear whether or it holds state.

Reply:
Edit: I now see that these mean different things and my prior comment is wrong.

I still think API is a meaningless word and would like to see it changed. useFormikApiState vs. useFormikApi vs. useFormik vs. useFormikState isn’t intuitive for end users.

Reply:
Formik API is the only word here that makes sense to me.

- A getter is not an action
- A method is a function belonging to an object in OOP, which isn't really applicable to functional programming like Formik whereas it might apply more in `react-hook-form` since they use classes under the hood.
- The type returned by `useFormik` is `FormikApi` and the type exposed by Context is also `FormikApi`, so `useFormikApi` seems to be the best naming convention for the context since we can't use `useFormikContext`.

If you do not believe that `useFormik` returns an API, that is a whole different story. I don't think it returns Methods, or Actions. Maybe Helpers?? But we already draw a distinction in TypeScript between Helpers, Handlers, and Config.

Let's decide what the TypeScript name for the return value of `useFormik` is, and we'll change the name of `useFormikApi` to match. Whatever it is, the Context holds that value so the names should match.

Reply:
I think it'll be super clear that "API" is the right word for the API that Formik provides if we document it separately from the various components and hooks that we provide, and separately from the State.

A person who simply uses `useFormik()` and builds everything else from scratch is only using the Formik API, and we should document that separately. The current documentation uses API Reference to mean "all the things" whereas the API Formik provides is just what is returned by `useFormik` and every other component or hook we provide is a shortcut for using that API.

Reply:
We can always break backwards compatibility and switch `useFormikContext` to just have the API, and then it'll be `useFormikContext` and `useFormikState`.

Reply:
I did the above, switched back to useFormikContext and useFormikState, abandoning backwards compatibility for clarity.